### ISSUE TITLE
JSON mapping Int* type errors

### ISSUE THREAD

**Issue by marceloboeira** [crystal-lang/crystal#2301]
Currently is not possible to use types other than `Int32` to represent integers.

For instance, that works perfectly:
```crystal
require "json"

class Foo
  JSON.mapping({ id: String, amount: Int32 })
end

response = "{\"id\": \"a32b78d97fc\", \"amount\": 5}"
instance = Foo.from_json(response)
pp instance.id
pp instance.amount
```
But replacing `Int32` with `Int8`:
```crystal
class Foo
  JSON.mapping({ id: String, amount: Int8 })
end
```

Leads to a `undefined method` error:

```
instantiating 'Int8:Class#new(JSON::PullParser)'
in /usr/local/Cellar/crystal-lang/0.13.0/src/libc.cr:103: undefined method 'to_i8' for JSON::PullParser
```

**Comment by asterite** [comment]
We can add them. I guess it would just read an int64 and then convert to the given int type... which might be a bit "dangerous".

Is there any reason you need an Int8 in your mapping? Is this value guaranteed in the JSON you are getting/sending?

**Comment by marceloboeira** [comment]
@asterite got it. 

I just thought that the `Int32` was not necessary on my field, since it has a limited range of values I would not like to spend 32 bits where I need only 8 or 16. I guess that should be available since the language supports all those kinds of variables since weird that they're not available everywhere.

**Comment by asterite** [comment]
@marceloboeira Makes sense. I'll add this soon :-)

**Comment by marceloboeira** [comment]
@asterite what is the catch ? I could give it a try on fixing this. I was looking at the code last night, I can try today.

**Comment by asterite** [comment]
A type just has to provide a constructor with `pull : JSON::PullParser`. For example this is for [Int32](https://github.com/crystal-lang/crystal/blob/master/src/json/from_json.cr#L44). For other int types it should be really similar.

We should also do the same for YAML. And add specs, of course :-)

**Comment by marceloboeira** [comment]
@asterite alright, I'll take a look as soon as possible.

**Comment by marceloboeira** [comment]
Solved at #2304
